java/data/FileWriter.java

```diff
@@ -0,0 +1,24 @@
+package data;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+
+public class FileWriter {
+    public void writeFile(String outputFile, List<String> linesChunk) {
+        try (BufferedWriter writer = Files.newBufferedWriter(Paths.get(outputFile), StandardOpenOption.CREATE, StandardOpenOption.APPEND)) {
```

Comment:
for SSD devices, this approach is acceptable.
for spindle disks, re-opening a file for append is a bad practice for large files, since the "seek" operation is O(n). so eventually writing N bytes (for a large enough N) would be O(N^2).

Reply:
Completely right!
Just completed the min heap solution, using it now as the default algorithm.